gemrb/core/DataFileMgr.h

```diff
@@ -38,13 +43,134 @@ namespace GemRB {
  * Abstract loader for .INI files
  */
 
+class KeyValueGroup {
+public:
+	using KeyValueMap = std::unordered_map<std::string, std::string>;
```

Comment:
Why use `std::unordered_map` over `StringMap`?

Reply:
That's so new I haven't yet realized what it does.

Reply:
I've changed that but the code now fails over this assertion that looks odd since lookup was negative but insertion wasn't done either:
https://github.com/gemrb/gemrb/blob/ec6f1282354625d5146e73fb77210d676342d6c8/gemrb/core/Strings/StringMap.h#L60-L62
Can continue tomorrow, hopefully.

Reply:
Interesting, what is `ins.first` pointing to?. It uses `CstrHashCI` and `CstrEqCI` to match nodes so let's examine where those might go wrong.

I see that `CstrHash` is erroneously terminating when encountering a '\0', but that would at worst generate hash collisions.

`CstrEqCI` is using `stricmp`, but should probably be using something else, otherwise equality will break for anything containing a '\0'. Also, `StringView` isn't guaranteed to be null terminated so that can explode. Does ASAN complain here for you?

Reply:
you can try rebasing on master

Reply:
That's better